src/doc/playback_tests.cpp

```diff
@@ -464,39 +505,361 @@ TEST(Playback, PingPongWithInnerReverse)
+          else if (B_Range[1] == lastFrame-1)
+            concat(expected, B_aniDir == doc::AniDir::PING_PONG ? pingPongSeq2[1] : pingPongSeq2[0]);
+          else if (B_Range[1] == lastFrame-2)
+            concat(expected, B_aniDir == doc::AniDir::PING_PONG ? pingPongSeq1[1] : pingPongSeq1[0]);
+
+          // Reproduce left side of the tag A
+          temp = goLeft({0, B_Range[0]-1});
+          concat(expected, temp);
+          temp = goRight({1, B_Range[0]-1});
+          concat(expected, temp);
+
+          // Tag B playback (only if tag B first frame doesn't match with the tag A first frame
+          if (B_Range[0] == 1)
+            concat(expected, B_aniDir == doc::AniDir::PING_PONG ? pingPongSeq2[0] : pingPongSeq2[1]);
+          else if (B_Range[0] == 2)
+            concat(expected, B_aniDir == doc::AniDir::PING_PONG ? pingPongSeq3[0] : pingPongSeq3[1]);
+
+
+          // Reproduce right side of the tag A
+          temp = goRight({B_Range[1]+1, lastFrame});
+          concat(expected, temp);
+          // Sequence end
+        }
+
+        // Test
+        Tag* tagA = make_tag("A", 0, 4, A_aniDir, 2);
+        Tag* tagB = make_tag("B", B_Range[0], B_Range[1], B_aniDir, 2);
+        auto sprite = make_sprite(lastFrame + 1, { tagA, tagB });
+        Playback play(sprite.get(), 0, Playback::Mode::PlayInLoop);
+
+        expect_frames(play, expected);
+        EXPECT_FALSE(play.isStopped());
+      }
+    }
+  }
+}
 
-  Tag* tagA = make_tag("A", 0, 4, AniDir::PING_PONG, 2);
-  Tag* tagB = make_tag("B", 1, 3, AniDir::PING_PONG_REVERSE, 3);
-  auto sprite = make_sprite(5, { tagA, tagB });
+TEST(Playback, OnePingPongInsideOther1Repeat)
+{
+  // A repeat = 1  ;  B repeat = 1
+  //
+  //     A            A            A
+  // *-------*    *-------*    *-------*
+  //   B              B              B
+  // *---*          *---*          *---*
+  // 0 1 2 3 4    0 1 2 3 4    0 1 2 3 4
+
+  const int lastFrame = 4;
+  std::vector<AniDir> A_AniDirs = {AniDir::PING_PONG, AniDir::PING_PONG_REVERSE};
+  std::vector<AniDir> B_AniDirs = {AniDir::PING_PONG, AniDir::PING_PONG_REVERSE};
+  std::vector<int> A_Range = {0,lastFrame};
+  std::vector<std::vector<int>> rangeBs = {{0,2}, {1,3}, {2,4}};
+  std::vector<std::vector<int>> pingPongSeq1 = {{0,1,2}, {2,1,0}};
+  std::vector<std::vector<int>> pingPongSeq2 = {{1,2,3}, {3,2,1}};
+  std::vector<std::vector<int>> pingPongSeq3 = {{2,3,4}, {4,3,2}};
+
+  for (auto A_aniDir : A_AniDirs) {
+    for (auto B_aniDir : B_AniDirs) {
+      for (auto B_Range : rangeBs) {
+        std::vector<int> expected;
+        std::vector<int> temp;
+        //     A            A            A
+        // <------->    <------->    <------->
+        //   B              B              B
+        // *---*          *---*          *---*
+        // 0 1 2 3 4    0 1 2 3 4    0 1 2 3 4
+        if (A_aniDir == doc::AniDir::PING_PONG) {
+
+          // Start
+          temp = goRight({0, B_Range[0]-1});
```

Comment:
Remove `{` and `}` from every `goRight()` and `goLeft()` call:
```suggestion
          temp = goRight(0, B_Range[0]-1);
```